### PR TITLE
Fix comment references, update lints, require latest Dart SDK

### PR DESCRIPTION
**Pull request by kevmoo**
Prepare to release v3.1.1

Also did cleanup to the markdown files

### CODE REVIEW

File: lib/src/percent.dart [hunk 1]
```diff
@@ -18,9 +18,8 @@ const percent = PercentCodec._();
 // TODO(nweiz): Add flags to support generating and interpreting "+" as a space
 // character. Also add an option for custom sets of unreserved characters.
 /// A codec that converts byte arrays to and from percent-encoded (also known as
-/// URL-encoded) strings according to [RFC 3986][rfc].
-///
-/// [rfc]: https://tools.ietf.org/html/rfc3986#section-2.1
+/// URL-encoded) strings according to
+/// [RFC 3986](https://tools.ietf.org/html/rfc3986#section-2.1).
```
Comment (kevmoo) [inline lib/src/percent.dart:22]: This is still a bug in the linter for comment references, sadly.

File: analysis_options.yaml [hunk 2]
```diff
@@ -1,45 +1,46 @@
...
-    - overridden_fields
+    - lines_longer_than_80_chars
+    - literal_only_boolean_expressions
+    - missing_whitespace_between_adjacent_strings
+    - no_adjacent_strings_in_list
+    - no_runtimeType_toString
+    - omit_local_variable_types
+    - only_throw_errors
     - package_api_docs
-    - package_names
```
Comment (kevmoo) [inline analysis_options.yaml:31]: these were all duplicated by lints v2

File: lib/src/fixed_datetime_formatter.dart [hunk 3]
```diff
@@ -25,12 +25,6 @@
 /// string, in this case `YYYY kiwi MM` is the same format string as
 /// `YYYY------MM`. When encoding a [DateTime], the non-format characters are in
 /// the output verbatim.
-///
-/// Note: this class differs from [DateFormat] in that here, the format
```
Comment (natebosch) [inline lib/src/fixed_datetime_formatter.dart:29]: This was useful context. Can we maybe link to the dartdoc manually?

Comment (kevmoo) [reply lib/src/fixed_datetime_formatter.dart:29]: WHERE is `DateFormat`? I couldn't find it!

Comment (kevmoo) [reply lib/src/fixed_datetime_formatter.dart:29]: In intl, maybe? I can look...

Comment (kevmoo) [reply lib/src/fixed_datetime_formatter.dart:29]: yup!

**Review by natebosch — APPROVED** [review]

**Pull request merged**